src/Rules/Functions/IncompatibleDefaultParameterTypeRule.php

```diff
@@ -60,7 +60,7 @@ public function processNode(Node $node, Scope $scope): array
 			$defaultValueType = $scope->getType($param->default);
 			$parameterType = $parameters->getParameters()[$paramI]->getType();
 
-			if ($parameterType->isSuperTypeOf($defaultValueType)->yes()) {
+			if ($parameterType->accepts($defaultValueType, true)->yes()) {
```

Comment:
The `accepts` semantics are different and generally shouldn't be used outside of "method call" and "function call" rules. For example they allow object with `__toString` be passed to `string`. It would also break with generics, as `T of X` generally doesn't accept any `X`.

Better solution is to detect when `parameterType` contains `float`. You can do that with:

```php
if ($parameterType->isSuperTypeOf(new FloatType())->yes()) {
    $parameterType = TypeCombinator::union($parameterType, new IntegerType());
}
```

Reply:
Builds are coming back green, so I guess these cases you mentioned need tests of their own. (Though, since it's impossible to have an object as default I'm not sure how they would be tested ...)

Reply:
I think that something like this would fail with your code:

```php
/**
 * @template T
 * @param T $x
 */
function foo($x = 1) {}
```

Reply:
I'll merge this if you use my suggestion and it doesn't break anything :)

Reply:
You're right. I'll add that to the test samples.